filament/backend/src/CommandBufferQueue.cpp

```diff
@@ -109,8 +109,8 @@ void CommandBufferQueue::flush() noexcept {
     }
 }
 
-std::vector<CommandBufferQueue::Slice> CommandBufferQueue::waitForCommands() const {
-    if (!UTILS_HAS_THREADING) {
+std::vector<CommandBufferQueue::Slice> CommandBufferQueue::waitForCommands(bool hasDedicatedDriverThread) const {
+    if (!UTILS_HAS_THREADING || !hasDedicatedDriverThread) {
```

Comment:
I think all usages of `UTILS_HAS_THREADING` need a much more careful review, a value of 1 longer implies the existence of a driver thread. (At least I see no indication in the PR description that this evaluation has happened). 

We have to identify one-by-one if a piece of code has the macro because it relies on the job system, having a driver thread, or both. On a quick glance these already look suspicious to me: `FRenderer::terminate()`, `FEngine::shutdown()`, `FilamentApp::run()`, `FFence::wait()`.

The best would be to be explicit about this everywhere, we can take advantage of the fact that it's enough to set it at compile-time. `UTILS_HAS_THREADING` could be turned into a three-state enum-like macro, or we could have a new two-state define (e.g. `FILAMENT_HAS_DRIVER_THREAD`, always set to zero if `UTILS_HAS_THREADING` is zero).

Reply:
https://github.com/shapr3d/filament/pull/104/commits/adb5195163044bc0e8a092ce94591e978af5ff82